### PR TITLE
Create module to make for easier use of PSC

### PR DESCRIPTION
**Pull request by daxmc99**
This adds a LICENSE along with a short example

I tested this in our internal test environment and it successfully created the PSC

### Test Plan

Applied to local test project

### TODO

- [ ] Make proxy network configurable

### CODE REVIEW

File: modules/gcp-passthrough-nlb/variables.tf [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    error_message = "must be a valid subnetwork name"
+  }
+}
+
+
+variable "name" {
+  default = "sourcegraph-psc"
+}
+
+variable "healthcheck-https-path" {
```
Comment (michaellzc) [inline modules/gcp-passthrough-nlb/variables.tf]:
```suggestion
variable "healthcheck_https_path" {
```

File: modules/gcp-passthrough-nlb/main.tf [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+  check_interval_sec  = 5
+  healthy_threshold   = 2
+  timeout_sec         = 5
+  unhealthy_threshold = 2
+
+  https_health_check {
+    host         = var.domain
+    proxy_header = "NONE"
+    request_path = var.healthcheck-https-path
```
Comment (michaellzc) [inline modules/gcp-passthrough-nlb/main.tf]:
```suggestion
    request_path = var.healthcheck_https_path
```

File: modules/gcp-passthrough-nlb/variables.tf [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+}
+
+variable "domain" {
+  type    = string
+  default = null
+}
+
+variable "domain_root" {
+  default = "sg.dev."
+}
```
Comment (michaellzc) [inline modules/gcp-passthrough-nlb/variables.tf]: let's remove this

File: modules/gcp-passthrough-nlb/variables.tf [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+  default = [
+    "sourcegraph-psc",
+  ]
+}
+
+variable "subnet_router_network_tags" {
+  default = [
+    "sourcegraph-psc",
+  ]
+}
```
Comment (michaellzc) [inline modules/gcp-passthrough-nlb/variables.tf]: let's remove this, it's not used

File: modules/gcp-passthrough-nlb/variables.tf [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+variable "healthcheck-https-path" {
+  default = "/status"
+}
+
+variable "domain" {
+  type    = string
+  default = null
+}
```
Comment (michaellzc) [inline modules/gcp-passthrough-nlb/variables.tf]: it looks like GHE does not care about Host header in order to make health check work, we can remove this?

**Comment by michaellzc** [comment]
another note:

let's remove all the `22` and `80` ports from firewall rules. they were there only for testing purposes, and they shouldn't be used in prod.

**Comment by daxmc99** [comment]
@michaellzc What about `ssh` cloning? We might want to leave port 22 open for that

**Comment by michaellzc** [comment]
> @michaellzc What about `ssh` cloning? We might want to leave port 22 open for that

I doubt anyone with GitHub will want to use ssh key for git clone.

Additionally, Cloud does not support configuring customer's SSH key today (it needs to be mounted from a CM or Secret to gitserver, and you can't do so from siteconfig)

File: modules/gcp-passthrough-nlb/main.tf [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+}
+
+
+terraform {
+  required_version = "> 1.2.3, < 2"
+
+  required_providers {
+    google = {
+      source  = "hashicorp/google"
+      version = "~> 4.79.0"
```
Comment (michaellzc) [inline modules/gcp-passthrough-nlb/main.tf]:
```suggestion
      version = ">= 4.79.0, < 5.0"
```

File: modules/gcp-passthrough-nlb/main.tf [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+    }
+
+    random = {
+      source  = "hashicorp/random"
+      version = "~> 3.5.1"
```
Comment (michaellzc) [inline modules/gcp-passthrough-nlb/main.tf]:
```suggestion
      version = ">= 3.5.1, < 4.0"
```

**Review by michaellzc — APPROVED** [review]

File: examples/gcp-passthrough-nlb/main.tf [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+  project = var.project_name
+  name    = var.subnet_name
+  region  = var.region
+}
+
+module "sourcegraph-psc" {
+  source = "../../modules/gcp-passthrough-nlb"
+
+  project_id = "sourcegraph-github-enterprise"
+  network_id = "ghe"
```
Comment (michaellzc) [inline examples/gcp-passthrough-nlb/main.tf:32]: why not also use a data source here? for consistency sake

File: examples/gcp-passthrough-nlb/main.tf [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+  compute_instance_name         = data.google_compute_instance.ghe.name
+  compute_instance_ip_address   = data.google_compute_instance.ghe.network_interface[0].network_ip
+  compute_instance_subnetwork   = data.google_compute_subnetwork.ghe.name
+  compute_instance_network_tags = data.google_compute_instance.ghe.tags
+
+  authorized_consumer_projects = {
+    "src-cd795a54649fc24d9b5f" : {
+      id : var.sourcegraph_provided_project_name,
+      limit : 10,
+    }
```
Comment (michaellzc) [inline examples/gcp-passthrough-nlb/main.tf]:
```suggestion
    "sourcegraph-psc" : {
      id : var.sourcegraph_provided_project_name,
      limit : 10,
    }
```

File: examples/gcp-passthrough-nlb/main.tf (hunk 8 above)
Comment (daxmc99) [reply examples/gcp-passthrough-nlb/main.tf:32]: Nice catch, updated

**Pull request merged**